pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/assembly/AssemblyVariantJsonServlet.java

```diff
@@ -201,7 +202,10 @@ protected boolean isValidResource(@Nonnull SlingHttpServletRequest request, @Non
                         .variants().get()
                         .canonicalVariant().get();
                 moduleMap.put("canonical_uuid", canonical.uuid().get());
-                moduleMap.put("title", page.title().get());
+                // Get the current title from module instead of using the stored value in the content node
+                // When module title is modified, the value stored in content node becomes stale.
+                // Sometime, the title stored in content node shows filename instead of title.
+                moduleMap.put("title", ServletHelper.getModuleTitleFromUuid(canonical));
```

Comment:
I won't block this, but I'll note that the name of the method is not good: it says `fromUuid` when in fact it doesn't accept a uuid, but a `ModuleVariant`.